OtripleS.Web.Api/Controllers/StudentExamFeesController.cs

```diff
@@ -0,0 +1,195 @@
+            }
+            catch (StudentExamFeeValidationException studentExamFeeValidationException)
+                when (studentExamFeeValidationException.InnerException is NotFoundStudentExamFeeException)
+            {
+                string innerMessage = GetInnerMessage(studentExamFeeValidationException);
+
+                return NotFound(innerMessage);
+            }
+            catch (StudentExamFeeValidationException studentExamFeeValidationException)
+            {
+                string innerMessage = GetInnerMessage(studentExamFeeValidationException);
+
+                return BadRequest(innerMessage);
+            }
+            catch (StudentExamFeeDependencyException studentExamFeeDependencyException)
+            {
+                return Problem(studentExamFeeDependencyException.Message);
+            }
+            catch (StudentExamFeeServiceException studentExamFeeServiceException)
+            {
+                return Problem(studentExamFeeServiceException.Message);
+            }
+        }
+
+        [HttpPut]
+        public async ValueTask<ActionResult<StudentExamFee>> PutStudentExamFeeAsync(StudentExamFee studentExamFee)
+        {
+            try
+            {
+                StudentExamFee registeredStudentExamFee =
+                    await this.studentExamFeeService.ModifyStudentExamFeeAsync(studentExamFee);
+
+                return Ok(registeredStudentExamFee);
+            }
+            catch (StudentExamFeeValidationException studentExamFeeValidationException)
+                when (studentExamFeeValidationException.InnerException is NotFoundStudentExamFeeException)
+            {
+                string innerMessage = GetInnerMessage(studentExamFeeValidationException);
+
+                return NotFound(innerMessage);
+            }
+            catch (StudentExamFeeValidationException studentExamFeeValidationException)
+            {
+                string innerMessage = GetInnerMessage(studentExamFeeValidationException);
+
+                return BadRequest(innerMessage);
+            }
+            catch (StudentExamFeeDependencyException studentExamFeeDependencyException)
+                when (studentExamFeeDependencyException.InnerException is LockedStudentExamFeeException)
+            {
+                string innerMessage = GetInnerMessage(studentExamFeeDependencyException);
+
+                return Locked(innerMessage);
+            }
+            catch (StudentExamFeeDependencyException studentExamFeeDependencyException)
+            {
+                return Problem(studentExamFeeDependencyException.Message);
+            }
+            catch (StudentExamFeeServiceException studentExamFeeServiceException)
+            {
+                return Problem(studentExamFeeServiceException.Message);
+            }
+        }
+
+        [HttpDelete("{studentExamFeeId}")]
+        public async ValueTask<ActionResult<StudentExamFee>> DeleteStudentExamFeeAsync(Guid studentExamFeeId)
```

Comment:
We are going to probably change that since we are looking based on a composite key.